### PR TITLE
RR-742 - Add 'Goals created successfully' message

### PR DESCRIPTION
**Pull request by nathanrussell-moj-digital**
This PR adds the success notification banner when a new goal(s) has been created:
Screenshot 2024-04-09 at 11 17 04

It also corrects something in my previous PR re: adding steps, in that the new empty step title field must be focussed when it is added.

### CODE REVIEW

File: integration_tests/e2e/goal/createGoals.cy.ts [hunk 1]
```diff
@@ -105,6 +105,7 @@ context('Create goals', () => {
     createGoalPage //
       .hasNoErrors()
       .goalHasNumberOfStepsFields(1, 2)
+      .stepTitleFieldIsFocussed(1, 2)
```
Comment (nathanrussell-moj-digital) [inline integration_tests/e2e/goal/createGoals.cy.ts:108]: Assert the correct step title field is focussed when a new step is added

File: integration_tests/e2e/goal/createGoals.cy.ts [hunk 2]
```diff
@@ -144,4 +145,31 @@ context('Create goals', () => {
...
+
+    // When
+    createGoalPage.submitPage()
+
+    // Then
+    overviewPage = Page.verifyOnPage(OverviewPage)
+    overviewPage //
+      .hasGoalsCreatedSuccessfullyMessage()
+      .refreshPage()
+      .doesNotHaveGoalsCreatedSuccessfullyMessage()
```
Comment (nathanrussell-moj-digital) [inline integration_tests/e2e/goal/createGoals.cy.ts:173]: Ensure the message is only displayed once, immediately after creating goals.
The starts on the Overview screen and asserts the message is not show. Then it creates a goal, arrives back on the Overview screen and asserts the message is displayed. Then the page is refreshed, and we assert the message is no longer displayed

File: server/views/pages/overview/index.njk [hunk 3]
```diff
@@ -19,6 +19,17 @@
 
 {% block content %}
 
+  {% if tab === 'overview' and showGoalCreationSuccessMessage %}
+    {{ mojBanner({
```
Comment (nathanrussell-moj-digital) [inline server/views/pages/overview/index.njk:23]: Personally (IMHO) I think the banner in the design looks a little ... errr ... meh! 🤷‍♂️ 
I think I'd rather see the regular GDS Notification banner, but what do I know? 🤣 

Our design team have assured me [this banner style is part of the MoJ Design System](https://design-patterns.service.justice.gov.uk/components/banner#success), so I guess its OK 🤷‍♂️

**Review by leejacobson-moj — APPROVED** [review]
Looks good. Approved. 👍

**Pull request merged**